VoodooInput/VoodooInputSimulator/VoodooInputSimulatorDevice.cpp

```diff
@@ -191,35 +191,42 @@ void VoodooInputSimulatorDevice::constructReportGated(const VoodooInputEvent& mu
         finger_data.Touch_Minor = 20;
         finger_data.Touch_Major = 20;
         
-        static bool hold_drag = false;
-        if (input_report.Button) {
-            input_report.Button = false;
-            finger_data.Pressure = 0xff;
-        } else if (!input_report.Button && multitouch_event.contact_count == 1) {
-            static IOFixed first_scaled_x = 0, first_scaled_y = 0;
-            static uint64_t press_start_ns;
-            uint64_t now_abs;
-            clock_get_uptime(&now_abs);
-            uint64_t now_ns;
-            absolutetime_to_nanoseconds(now_abs, &now_ns);
-            if (touch_state[i] == 1) {
-                if (transducer->isTransducerActive) {
-                    first_scaled_x = scaled_x;
-                    first_scaled_y = scaled_y;
-                    press_start_ns = now_ns;
-                    hold_drag = true;
-                } else {
-                    hold_drag = false;
-                }
-            } else if (press_start_ns && hold_drag && now_ns - press_start_ns < 500e6) {
-                if (abs(scaled_x - first_scaled_x) > MT2_MAX_X / 50 || abs(scaled_y - first_scaled_y) > MT2_MAX_Y / 50) {
-                    hold_drag = false;
-                }
-            } else if (hold_drag) {
-                finger_data.Pressure = 100;
-            }
+        if (transducer->supportsPressure) {
+            finger_data.Pressure = transducer->currentCoordinates.pressure;
+            finger_data.Size = transducer->currentCoordinates.width;
+            finger_data.Touch_Major = transducer->currentCoordinates.width;
+            finger_data.Touch_Minor = transducer->currentCoordinates.width;
         } else {
-            hold_drag = false;
+            if (touch_state[i] > 4) {
```

Comment:
Ideally, we shouldn't be keeping track of how long the touch input has been pressed down. This should be handled by Apple's own MT stack. Hence the reason why I reverted back to the old way of spoofing pressure.

Reply:
Actually, the mentioned snippet  from the commit https://github.com/Goshin/VoodooI2C/commit/d8c98a84b274d893c2ad74a6c7f7331ae561474f is for a personal custom feature, which is in the `master` branch of my fork repo. The commits related to the PR https://github.com/alexandred/VoodooI2C/pull/254 are in the `impv-mt` branch.

Reply:
I suggest that we leave it out as it makes taps not properly register (e.g. two finger taps do not register as a right click).

Reply:
I just meant you might merge the wrong branch which has some unrelated commits.

Reply:
You are indeed correct, I must've been looking at your local master.

Reply:
@Goshin can you double check that the port is good to go?

Reply:
looks good to me.